mlflow/utils/docstring_utils.py

```diff
@@ -285,9 +285,23 @@ class that describes the model's inputs and outputs. If not specified but an
 imports declared from a common root path if multiple files are defined with import dependencies
 between them to avoid import errors when loading the model.
 
+You can leave ``code_paths`` argument unset but set ``infer_code_paths`` to ``True`` to let MLflow
+infer the model code paths. See ``infer_code_paths`` argument doc for details.
+
 For a detailed explanation of ``code_paths`` functionality, recommended usage patterns and
 limitations, see the
 `code_paths usage guide <https://mlflow.org/docs/latest/model/dependencies.html?highlight=code_paths#saving-extra-code-with-an-mlflow-model>`_.
+"""
+        ),
+        "infer_code_paths": (
+            """If set to ``True``, MLflow automatically infers model code paths. The inferred code path files only
+            includes necessary python module files. Only python code files under current working directory
+            is automatic inferrable. Default value is ``False``.
+
+.. warning::
+    Please ensure that the custom python module code does not contain sensitive data such as
+    credential token strings, otherwise they might be included in the automatic inferred code path files
```

Comment:
Any other scenarios where this doesn't work? If any, I think we should also clarify them.

Reply:
and this argument must be experimental. I feel like there are many unknown corner cases that the inference doesn't work.

Reply:
> Any other scenarios where infer_code_paths doesn't work or infer_code_paths should not be used? 

So far I can only think out the limitation of inferring current directory files.

Reply:
what if your custom python script depends on  a json file?

Reply:
> what if your custom python script depends on a json file?

oh yes this is another case. let me update doc.

Reply:
@WeichenXu123 It might not be a config.

Reply:
Yes I update doc saying it as resource files